x.py fix is broken

If you try to run x.py fix you'll be getting an error:

```
$ ./x.py fix 
Building bootstrap
    Finished dev [unoptimized] target(s) in 0.09s
Checking stage0 library artifacts (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
    Checking test v0.0.0 (/rust/library/test)
error[E0463]: can't find crate for `test`
 --> library/test/src/stats/tests.rs:3:1
  |
3 | extern crate test;
  | ^^^^^^^^^^^^^^^^^^ can't find crate

error[E0463]: can't find crate for `test`
    --> library/test/src/stats/tests.rs:43:1
     |
42   |   #[test]
     |   ------- in this procedural macro expansion
43   | / fn test_min_max_nan() {
44   | |     let xs = &[1.0, 2.0, f64::NAN, 3.0, 4.0];
45   | |     let summary = Summary::new(xs);
46   | |     assert_eq!(summary.min, 1.0);
47   | |     assert_eq!(summary.max, 4.0);
48   | | }
     | |_^ can't find crate
     |
    ::: library/core/src/macros/mod.rs:1495:5
     |
1495 |       pub macro test($item:item) {
     |       -------------- in this expansion of `#[test]`

[tons of repetitions of this error for each test in the test crate]
```

@rustbot label T-bootstrap

--- a/src/bootstrap/check.rs
+++ b/src/bootstrap/check.rs
@@ -99,6 +99,10 @@ impl Step for Std {
             cargo_subcommand(builder.kind),
         );
         std_cargo(builder, target, compiler.stage, &mut cargo);
+        if matches!(builder.config.cmd, Subcommand::Fix { .. }) {
+            // By default, cargo tries to fix all targets. Tell it not to fix tests until we've added `test` to the sysroot.
+            cargo.arg("--lib");
+        }
 
         builder.info(&format!(
             "Checking stage{} library artifacts ({} -> {})",

